### PR TITLE
Add option for renaming files in data_grabber

### PR DESCRIPTION
**Pull request by wbarnha**
I wanted to add a feature to data_grabber.py so that I could rename the files by right-clicking on them rather than going to manually the plot legend.

Also some other changes are simple syntax reformatting.

### CODE REVIEW

**Comment by wbarnha** [comment]
Apparently you could do this by just double-clicking the file name, this never occurred to me until now. Here's the code anyway if you still want it.

**Comment by jhillairet** [comment]
seems OK to me, but eventually @dvincentwest could review as he is the author of the qt widget code.

**Comment by dvincentwest** [comment]
I haven't spent a lot of time dealing with PyQt for almost 3 years now, so unfortunately I can't give you much direction.  I tested the changes, and they work, and I think that having a right click option is a good idea.  I think also adding an "F2" key press which is probably a Qt named constant somewhere would also be a good idea.

I think it would be better though if you can figure out how to route the editing through the standard QListWidgetItem ItemDelegate editor, which is what happens when you double click on the item currently.  Right now you pop up a separate line editor on top that is narrower and also offset which looks a little distracting, and also that you can't escape out of to cancel your edit.  Once you hit the right-clicked option, your only way out is to accept your changes, but sometimes you don't want to do that.

Basically, I think what you want is for the right click action to do the same thing that the double click action does, not a different thing.  I don't know how to do that though.  Been too long

**Comment by wbarnha** [comment]
Got the requested change to have double click editing to be the same for right click rename. Apparently all you need to do is:
```
    def rename_item(self):
        item = self.currentItem()
        item.setFlags(item.flags() | QtCore.Qt.ItemIsEditable)
        self.editItem(item)
```

**Review by dvincentwest — APPROVED** [review]
I tested and the changes behave as expected

**Comment by jhillairet** [comment]
> I tested and the changes behave as expected

OK, thank you.

**Comment by jhillairet** [comment]
> Got the requested change to have double click editing to be the same for right click rename. Apparently all you need to do is:

@wbarnha is it OK to merge?

**Comment by wbarnha** [comment]
Yeah, you can go ahead and merge now.

**Comment by jhillairet** [comment]
Thank you.
